www/js/diary/list.js

```diff
@@ -766,10 +766,20 @@ angular.module('emission.main.diary.list',['ui-leaflet',
           if (angular.isDefined(Timeline.data.currDay)) {
               // page was already loaded, reload it automatically
               readAndUpdateForDay(Timeline.data.currDay);
+
+              const tripProp = notificationData.properties;
+              const trip = $scope.data.currDayTripWrappers.find(function(tripgj, index, array) {
+                return (
+                  tripgj.properties.start_ts === tripProp.start_ts &&
+                  tripgj.properties.end_ts === tripProp.end_ts
+                );
+              });
+              $scope.confirmSurvey(trip);
```

Comment:
Not sure if this is what you are suggesting...

Reply:
Yeah, that works. Since you know that this is a trip-end prompt, I think that a simpler and more performant option would be to just use the last trip (e.g. sth like `$scope.data.currDayTripWrappers[$scope.data.currDayTripWrappers.length - 1]`

Reply:
Since Timeline is a service, you could also call `Timeline.updateForDay(day);` from the `post-trip-prompt.js` directly. But I think that after the survey you want to return to the diary, so this is probably the better option after all.

Reply:
> Yeah, that works. Since you know that this is a trip-end prompt, I think that a simpler and more performant option would be to just use the last trip (e.g. sth like `$scope.data.currDayTripWrappers[$scope.data.currDayTripWrappers.length - 1]`

Just to be cautious, sometimes users has multiple trips in a short period and got multiple notifications. This optimization will definitely cause the problem.